src/init.cpp

```diff
@@ -1628,7 +1628,7 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     // Either install a handler to notify us when genesis activates, or set fHaveGenesis directly.
     // No locking, as this happens before any background thread is started.
     boost::signals2::connection block_notify_genesis_wait_connection;
-    if (chainman.ActiveChain().Tip() == nullptr) {
+    if (WITH_LOCK(::cs_main, return chainman.ActiveChain().Tip() == nullptr)) {
```

Comment:
Nevermind, I see the WITH_LOCK. mb

Reply:
Calling `Tip()` will need the mutex, as it accesses the underlying `std:vector`. Using a `CBlockIndex*` may or may not need the mutex, so I think it it fine to request the caller to figure that out.

Reply:
Should the `CChain` functions that use `vChain` have the exclusive lock annotation?

Reply:
I am not sure if we want to add validation mutex lock annotations to classes that are only supposed to be primitive data structures.

Maybe for now this could make sense to document the status quo. However, I'd like to postpone this to the future.

Alternatively, we could start thinking about adding a CChain member mutex, similar to the mempool one?

Reply:
I think adding a member mutex would be a good idea in the future